### PR TITLE
[wip] test/extended/dr: DR with existent nodes

### PR DESCRIPTION
**Pull request by hexfusion**
currently, DR tests recovery by creating new machines, we have found a bug in scaling with existent nodes this PR is to automate that testing.

### CODE REVIEW

**Comment by hexfusion** [comment]
Probably hit the DR failure condition on latest

/test e2e-aws-disruptive

**Comment by hexfusion** [comment]
same https://prow.svc.ci.openshift.org/view/gcs/origin-ci-test/pr-logs/pull/25172/pull-ci-openshift-origin-master-e2e-aws-disruptive/1274493456856649728

/test e2e-aws-disruptive

**Comment by hexfusion** [comment]
Decent failure percentage, both failures appear to be same condition.
3 pass 
2 fail
https://deck-ci.apps.ci.l2s4.p1.openshiftapps.com/pr-history/?org=openshift&repo=origin&pr=25172

**Comment by hexfusion** [comment]
more data 
/test e2e-aws-disruptive

File: test/extended/dr/quorum_restore.go [hunk 1]
```diff
@@ -92,63 +83,37 @@ var _ = g.Describe("[sig-etcd][Feature:DisasterRecovery][Disruptive]", func() {
...
 					}
+					framework.Logf("Stopping etcd and kube-apiserver pods and removing data-dir from %s", node.Name)
+					expectSSH("sudo -i /bin/bash -cx 'mv /etc/kubernetes/manifests/etcd-pod.yaml /tmp'", node)
 
-					framework.Logf("Destroying %s", masterMachine)
-					err = ms.Delete(context.Background(), masterMachine, metav1.DeleteOptions{})
-					o.Expect(err).NotTo(o.HaveOccurred())
+					time.Sleep(180 * time.Second)
+
+					expectSSH("sudo -i /bin/bash -cx 'mv /etc/kubernetes/manifests/kube-apiserver-pod.yaml /tmp; rm -rf /var/lib/etcd'", node)
```
Comment (tedyu) [inline test/extended/dr/quorum_restore.go:106]: kube-apiserver takes ~70 seconds to shutdown.
Should there be some waiting after the manifest file is moved ?

Comment (hexfusion) [reply test/extended/dr/quorum_restore.go:106]: in our tests, we found reversing order created another failure case that required kubelet to restart to un wedge. I will add same wait afterward as that does make sense. I will test the ordering change locally.

**Comment by hexfusion** [comment]
infra flake 

/test e2e-aws-disruptive

File: test/extended/dr/quorum_restore.go (hunk 1 above)
Comment (hexfusion) [reply test/extended/dr/quorum_restore.go:106]: @tedyu first local run of this test with etcd and apiserver reversed order failed. Signature is a little different. New Pods are in `Pending` status indefinitely.

```NAME                                                           READY   STATUS      RESTARTS   AGE
etcd-ip-10-0-129-105.us-west-1.compute.internal                1/1     Running     0          26m
etcd-ip-10-0-135-219.us-west-1.compute.internal                4/4     Running     1          103m
installer-2-ip-10-0-129-105.us-west-1.compute.internal         0/1     Completed   0          102m
installer-2-ip-10-0-135-219.us-west-1.compute.internal         0/1     Completed   0          103m
installer-2-ip-10-0-203-76.us-west-1.compute.internal          0/1     Completed   0          103m
installer-3-ip-10-0-203-76.us-west-1.compute.internal          0/1     Pending     0          25m
revision-pruner-2-ip-10-0-129-105.us-west-1.compute.internal   0/1     Completed   0          99m
revision-pruner-2-ip-10-0-135-219.us-west-1.compute.internal   0/1     Completed   0          99m
revision-pruner-2-ip-10-0-203-76.us-west-1.compute.internal    0/1     Completed   0          99
```

```
installer-2-ip-10-0-203-76.us-west-1.compute.internal          0/1     Completed   0          104m
installer-3-ip-10-0-129-105.us-west-1.compute.internal         0/1     Completed   0          103m
installer-3-ip-10-0-135-219.us-west-1.compute.internal         0/1     Completed   0          103m
installer-3-ip-10-0-203-76.us-west-1.compute.internal          0/1     Completed   0          100m
installer-4-ip-10-0-203-76.us-west-1.compute.internal          0/1     Completed   0          99m
installer-5-ip-10-0-129-105.us-west-1.compute.internal         0/1     Completed   0          96m
installer-5-ip-10-0-203-76.us-west-1.compute.internal          0/1     Completed   0          98m
installer-6-ip-10-0-129-105.us-west-1.compute.internal         0/1     Completed   0          94m
installer-6-ip-10-0-135-219.us-west-1.compute.internal         0/1     Completed   0          92m
installer-7-ip-10-0-129-105.us-west-1.compute.internal         0/1     Completed   0          83m
installer-7-ip-10-0-135-219.us-west-1.compute.internal         0/1     Completed   0          91m
installer-7-ip-10-0-203-76.us-west-1.compute.internal          0/1     Completed   0          87m
installer-8-ip-10-0-203-76.us-west-1.compute.internal          0/1     Pending     0          26m
kube-apiserver-ip-10-0-129-105.us-west-1.compute.internal      4/4     Running     6          83m
kube-apiserver-ip-10-0-135-219.us-west-1.compute.internal      4/4     Running     0          91m
kube-apiserver-ip-10-0-203-76.us-west-1.compute.internal       4/4     Running     0          87m
revision-pruner-2-ip-10-0-203-76.us-west-1.compute.internal    0/1     Completed   0          103m
revision-pruner-3-ip-10-0-129-105.us-west-1.compute.internal   0/1     Completed   0          103m
revision-pruner-3-ip-10-0-135-219.us-west-1.compute.internal   0/1     Completed   0          100m
revision-pruner-3-ip-10-0-203-76.us-west-1.compute.internal    0/1     Completed   0          100m
revision-pruner-4-ip-10-0-203-76.us-west-1.compute.internal    0/1     Completed   0          98m
revision-pruner-5-ip-10-0-129-105.us-west-1.compute.internal   0/1     Completed   0          94m
revision-pruner-5-ip-10-0-203-76.us-west-1.compute.internal    0/1     Completed   0          97m
revision-pruner-6-ip-10-0-129-105.us-west-1.compute.internal   0/1     Completed   0          92m
revision-pruner-6-ip-10-0-135-219.us-west-1.compute.internal   0/1     Completed   0          91m
revision-pruner-7-ip-10-0-129-105.us-west-1.compute.internal   0/1     Completed   0          79m
revision-pruner-7-ip-10-0-135-219.us-west-1.compute.internal   0/1     Completed   0          87m
revision-pruner-7-ip-10-0-203-76.us-west-1.compute.internal    0/1     Completed   0          83m
```

Comment (hexfusion) [reply test/extended/dr/quorum_restore.go:106]: So what is interesting here is this pod is bogus.

> etcd-ip-10-0-135-219.us-west-1.compute.internal                4/4     Running     1          103m

If I delete just this single mirror pod everything completes as expected.

**Comment by hexfusion** [comment]
@tedyu first local run of this test with etcd and apiserver reversed order failed. Signature is a little different. New Pods are in `Pending` status indefinitely.

```NAME                                                           READY   STATUS      RESTARTS   AGE
etcd-ip-10-0-129-105.us-west-1.compute.internal                1/1     Running     0          26m
etcd-ip-10-0-135-219.us-west-1.compute.internal                4/4     Running     1          103m
installer-2-ip-10-0-129-105.us-west-1.compute.internal         0/1     Completed   0          102m
installer-2-ip-10-0-135-219.us-west-1.compute.internal         0/1     Completed   0          103m
installer-2-ip-10-0-203-76.us-west-1.compute.internal          0/1     Completed   0          103m
installer-3-ip-10-0-203-76.us-west-1.compute.internal          0/1     Pending     0          25m
revision-pruner-2-ip-10-0-129-105.us-west-1.compute.internal   0/1     Completed   0          99m
revision-pruner-2-ip-10-0-135-219.us-west-1.compute.internal   0/1     Completed   0          99m
revision-pruner-2-ip-10-0-203-76.us-west-1.compute.internal    0/1     Completed   0          99
```

```
installer-2-ip-10-0-203-76.us-west-1.compute.internal          0/1     Completed   0          104m
installer-3-ip-10-0-129-105.us-west-1.compute.internal         0/1     Completed   0          103m
installer-3-ip-10-0-135-219.us-west-1.compute.internal         0/1     Completed   0          103m
installer-3-ip-10-0-203-76.us-west-1.compute.internal          0/1     Completed   0          100m
installer-4-ip-10-0-203-76.us-west-1.compute.internal          0/1     Completed   0          99m
installer-5-ip-10-0-129-105.us-west-1.compute.internal         0/1     Completed   0          96m
installer-5-ip-10-0-203-76.us-west-1.compute.internal          0/1     Completed   0          98m
installer-6-ip-10-0-129-105.us-west-1.compute.internal         0/1     Completed   0          94m
installer-6-ip-10-0-135-219.us-west-1.compute.internal         0/1     Completed   0          92m
installer-7-ip-10-0-129-105.us-west-1.compute.internal         0/1     Completed   0          83m
installer-7-ip-10-0-135-219.us-west-1.compute.internal         0/1     Completed   0          91m
installer-7-ip-10-0-203-76.us-west-1.compute.internal          0/1     Completed   0          87m
installer-8-ip-10-0-203-76.us-west-1.compute.internal          0/1     Pending     0          26m
kube-apiserver-ip-10-0-129-105.us-west-1.compute.internal      4/4     Running     6          83m
kube-apiserver-ip-10-0-135-219.us-west-1.compute.internal      4/4     Running     0          91m
kube-apiserver-ip-10-0-203-76.us-west-1.compute.internal       4/4     Running     0          87m
revision-pruner-2-ip-10-0-203-76.us-west-1.compute.internal    0/1     Completed   0          103m
revision-pruner-3-ip-10-0-129-105.us-west-1.compute.internal   0/1     Completed   0          103m
revision-pruner-3-ip-10-0-135-219.us-west-1.compute.internal   0/1     Completed   0          100m
revision-pruner-3-ip-10-0-203-76.us-west-1.compute.internal    0/1     Completed   0          100m
revision-pruner-4-ip-10-0-203-76.us-west-1.compute.internal    0/1     Completed   0          98m
revision-pruner-5-ip-10-0-129-105.us-west-1.compute.internal   0/1     Completed   0          94m
revision-pruner-5-ip-10-0-203-76.us-west-1.compute.internal    0/1     Completed   0          97m
revision-pruner-6-ip-10-0-129-105.us-west-1.compute.internal   0/1     Completed   0          92m
revision-pruner-6-ip-10-0-135-219.us-west-1.compute.internal   0/1     Completed   0          91m
revision-pruner-7-ip-10-0-129-105.us-west-1.compute.internal   0/1     Completed   0          79m
revision-pruner-7-ip-10-0-135-219.us-west-1.compute.internal   0/1     Completed   0          87m
revision-pruner-7-ip-10-0-203-76.us-west-1.compute.internal    0/1     Completed   0          83m
```

File: test/extended/dr/quorum_restore.go (hunk 1 above)
Comment (hexfusion) [reply test/extended/dr/quorum_restore.go:106]: moving comments to main thread

**Comment by hexfusion** [comment]
So what is interesting here is this pod is bogus.

> etcd-ip-10-0-135-219.us-west-1.compute.internal                4/4     Running     1          103m

If I delete just this single mirror pod everything completes as expected.

**Comment by hexfusion** [comment]
[https://coreos.slack.com/files/UDVU5F0TY/F015E66MFP1/crio_service.log](url)

**Comment by hexfusion** [comment]
last disruptive test net failed but DR test passed.

/test e2e-aws-disruptive

**Comment by hexfusion** [comment]
flake 
/test e2e-aws-disruptive

**Comment by hexfusion** [comment]
passed
/test e2e-aws-disruptive

**Comment by hexfusion** [comment]
fail
/test e2e-aws-disruptive

**Pull request closed without merging**